File: TakeFive/takeFive/src/main/java/com/teuskim/takefive/common/MiscUtil.java
package com.teuskim.takefive.common;

import android.content.res.Resources;

public class MiscUtil {

	private static float dpToPxRatio = -1;

	/**
	 * DP 값을 PX 값으로 변환
	 */
	public static int convertDpToPx(float dp, Resources r) {
		if (dpToPxRatio < 0) {
			dpToPxRatio = r.getDisplayMetrics().density;
		}

		int px = Math.round(dp * dpToPxRatio);
		return px;
	}
}

File: TakeFive_kotlin/takeFive/src/main/java/com/teuskim/takefive/common/MiscUtil.kt
package com.teuskim.takefive.common

import android.content.res.Resources
import com.teuskim.takefive.common.MiscUtil.dpToPxRatio

object MiscUtil {

    private var dpToPxRatio = -1f

    /**
     * DP 값을 PX 값으로 변환
     */
    fun convertDpToPx(dp: Float, r: Resources): Int {
        if (dpToPxRatio < 0) {
            dpToPxRatio = r.displayMetrics.density
        }

        return Math.round(dp * dpToPxRatio)
    }
}
